### PR TITLE
Add HTTP endpoint for breakdown of log messages based on Kubernetes control plane component. 

### PR DESCRIPTION
**Pull request by AmartC**
This pull request adds an HTTP endpoint control_plane which will return the breakdown of Normal, Suspicious and Anomalous log messages based on the Kubernetes control plane component. The README has also been updated to reflect this change. To test out this Pull Request, you can use the following image: amartyarancher/opni-insights-service-amartya-test and test on a Kubernetes cluster which is able to send over control plane logs (RKE1, RKE2, K3S).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Opni Insights service
 
-* This repository builds the opni-insights image which sets up an HTTP server that will return the breakdown of normal, suspicious and anomolous log messages by pod, namespace and workload between a specified time interval provided in nanoseconds. This server can also return the overall breakdown of normal, suspicious and anomalous log messages within a specified time interval by specifying a Get request to the overall_insights endpoint. It can also return the suspicious and anomalous log messages within a specified time interval by submitting a Get request to the logs endpoint.
+* This repository builds the opni-insights image which sets up an HTTP server. It sets up endpoints for many different insights between any time interval.
+    * Breakdown of normal, suspicious and anomalous log message by pod through the pod endpoint
```
Comment (tybalex) [inline README.md:4]: we can probably summarize line 4,5,6,7 with `breakdown of normal, suspicious and anomalous log messages by pod/namespace/workload/control-plane-component`?

Comment (AmartC) [reply README.md:4]: Maybe, I just want it to be clear which endpoint the user should submit a Get request to.

**Review by tybalex — DISMISSED** [review]

File: opni-insights-service/app/main.py [hunk 2]
```diff
@@ -333,6 +333,59 @@ async def get_logs(start_ts, end_ts):
...
+        "kube-api-server",
+        "kube-proxy",
+        "kube-scheduler",
+        "etcd",
+        "k3s-agent",
+        "k3s-server",
+        "rke2-agent",
+        "rke2-server",
+    ]
+    for component_name in kubernetes_components:
```
Comment (dbason) [inline opni-insights-service/app/main.py:352]: Same feedback here, we should be using elasticsearch aggregations

**Review by dbason — DISMISSED** [review]

File: opni-insights-service/app/main.py (hunk 2 above)
Comment (sanjay920) [reply opni-insights-service/app/main.py:352]: @AmartC I second this. Even though it's a first pass aggregations are easy to do and are **way** more efficient

File: opni-insights-service/app/main.py [hunk 3]
```diff
@@ -333,6 +333,59 @@ async def get_logs(start_ts, end_ts):
...
+        try:
+            component_dict = {
+                "Name": component_name,
+                "Insights": {"Normal": 0, "Suspicious": 0, "Anomaly": 0},
```
Comment (sanjay920) [inline opni-insights-service/app/main.py:356]: Pls refactor to use ES aggregations

**Review by sanjay920 — CHANGES_REQUESTED** [review]

**Review by sanjay920 — APPROVED** [review]
If we are looking back in time, we can't receive pod/namespace/workload insights because we are relying on the Kubernetes API. We should not return pods insights in this manner because we have that metadata encoded in every log message + we should build up a historical lookup table in which we can provide these insights even if the request is in the past

**Review by dbason — APPROVED** [review]

**Pull request merged**